### PR TITLE
Add payment calculator

### PR DESCRIPTION
**Pull request by timabell**
### Context

This was previously a separate repository but the wider team have agreed that we don't currently have a need to split this out into separate services.

### Changes proposed in this pull request

1. Add payment calculator in separate directory (for clear history of the move)
2. Move everything to the right places in another commit

This is a copy paste from https://github.com/DFE-Digital/cpd-payment-calculations/tree/fd30b3ed074cb9029126963984b49452fe38459d

### Guidance to review

See the new specs mingled with some existing specs with:

```bash
bundle exec rspec -fd spec/lib spec/features
```

### CODE REVIEW

**Comment by timabell** [comment]
fixed linter error and rebased

File: documentation/adr/0001-record-architecture-decisions.md
Comment (jamesrose) [inline documentation/adr/0001-record-architecture-decisions.md:1]: Should these be in https://github.com/DFE-Digital/cpd-adr?

File: lib/services/ecf/output_payment_aggregator.rb
Comment (jamesrose) [inline lib/services/ecf/output_payment_aggregator.rb:3]:
```suggestion
module PaymentCalculator
```

Can we update the namespace to something that better describes this area?

File: spec/spec_helper.rb
Comment (jamesrose) [inline spec/spec_helper.rb:25]: I don't think this warrants a gem. Can we autoload these?

File: lib/meta.rb
Comment (jamesrose) [inline lib/meta.rb:3]: Should this be in the calculator namespace?

File: lib/initialize_with_config.rb
Comment (jamesrose) [inline lib/initialize_with_config.rb:7]: Should this be in the calculator namespace?

File: documentation/adr/0001-record-architecture-decisions.md
Comment (timabell) [reply documentation/adr/0001-record-architecture-decisions.md:1]: I don't think we're going to be using that repo.... @mattonfoot ?

File: lib/meta.rb
Comment (timabell) [reply lib/meta.rb:3]: I think @blackrat wanted to delete it entirely

File: lib/initialize_with_config.rb
Comment (timabell) [reply lib/initialize_with_config.rb:7]: Seems sensible now it's part of a bigger thing

File: lib/meta.rb
Comment (blackrat) [reply lib/meta.rb]: Yes. `InitializeWithConfig` has been changed to make Meta unnecessary.

File: lib/initialize_with_config.rb
Comment (blackrat) [reply lib/initialize_with_config.rb:6]: It's not necessarily just for the calculator. I'd saw leave it in lib.

File: lib/services/ecf/output_payment_aggregator.rb
Comment (timabell) [reply lib/services/ecf/output_payment_aggregator.rb]: done, good call

Comment (timabell) [reply lib/services/ecf/output_payment_aggregator.rb]: (specs still to do)

File: spec/spec_helper.rb
Comment (blackrat) [reply spec/spec_helper.rb]: @jamesrose Hmm. Do you mean autoload?  Wouldn't it be better to require them explicitly if we don't like that testing gem. 

Setting the config.autoload_paths array seems a little heavy? I'm not even sure that's documented in the file anymore. I've got some experience of doing that, and it's normally quite hard for people to find unless they know how Rails internals work.

Comment (blackrat) [reply spec/spec_helper.rb]: Or do you mean to do it at the application level the same way that it's being done to force reloading of decorators and serializers? Scoped for test in application.rb?

Comment (blackrat) [reply spec/spec_helper.rb]: Ah. Lightbulb moment. It's just for Rspec, so I'm guessing you can do that just in the .rspec file.

Comment (timabell) [reply spec/spec_helper.rb]: @jamesrose Are you happier with 3a593d9ec88a4954e1afe7b3399cb9d87336c8ef by @blackrat that eliminates it?
It does mean that there are lots more require statements around, which maybe is fine.

Comment (jamesrose) [reply spec/spec_helper.rb]: @timabell @blackrat Yes, I think that's fine 👍

**Review by jamesrose — APPROVED** [review]
Nice work getting this in - one step closer to an E2E calculation. 🚢

File: spec/spec_helper.rb
Comment (timabell) [inline spec/spec_helper.rb:16]: I'm not yet sure where the problem is, but we seem to now be running the cypress features with rspec which obviously doesn't work. They are then marked as pending with "no such step" hence not failing on CI :worried: 
This was introduced in the first two commits of the branch. Currently tracking this down. Don't merge till this is fixed.

Comment (timabell) [reply spec/spec_helper.rb:16]: Okay I think I see the problem, turnip seems to be hard-coded to load all feature files and is picking up the cypress feature files. Dang.

https://github.com/jnicklas/turnip/blob/00bd963dea49c5ff8f87edb1bcafcdb73f310c6a/lib/turnip/rspec.rb#L124

```ruby
  config.pattern += ',**/*.feature'
```

**Review by blackrat — CHANGES_REQUESTED** [review]
Turnip gem is incorrectly picking up the cypress tests, so rejecting this to give us a chance to fix this before someone clicks merge.

File: spec/spec_helper.rb
Comment (timabell) [reply spec/spec_helper.rb:16]: Fixed by adding and configuring `rutabaga` gem instead of directly using `turnip`.

File: lib/services/ecf/output_payment_aggregator.rb
Comment (timabell) [reply lib/services/ecf/output_payment_aggregator.rb]: specs done

**Review by blackrat — APPROVED** [review]

**Pull request merged**
